OloEngine/tests/DialogueTreeSerializationTest.cpp

```
@@ -0,0 +1,234 @@
+
+    // Verify response node text
+    auto* response = deserialized->FindNode(UUID(1003));
+    ASSERT_NE(response, nullptr);
+    auto textIt = response->Properties.find("text");
+    ASSERT_NE(textIt, response->Properties.end());
+    EXPECT_EQ(std::get<std::string>(textIt->second), "Move along.");
+
+    // Verify connections (order-independent)
+    ASSERT_EQ(deserialized->GetConnections().size(), 2u);
+    const auto& conns = deserialized->GetConnections();
+    bool hasOutputConn = std::ranges::any_of(conns, [](const auto& c)
```

Clang (14.0.6)

[warning] variable 'hasOutputConn' is not initialized

(cppcoreguidelines-init-variables)